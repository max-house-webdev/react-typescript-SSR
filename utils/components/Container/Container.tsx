import React from 'react';

export interface IContainerProps {
  As?: 'div' | 'header' | 'main' | 'footer' | 'section';
  className?: string;
  children?: React.ReactNode;
}

export function Container({
  As = 'div',
  className,
  children,
}: IContainerProps) {
  return <As className={className}>{children}</As>;
}
