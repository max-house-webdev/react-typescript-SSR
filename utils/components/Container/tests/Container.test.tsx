import React from 'react';
import { shallow } from 'enzyme';

import { Container } from '../Container';

describe('<Container />', () => {
  describe('should render as', () => {
    test('div by default', () => {
      const wrapper = shallow(<Container children={<div />} />);

      expect(wrapper).toMatchSnapshot();
    });

    test('header', () => {
      const wrapper = shallow(<Container As={'header'} children={<div />} />);

      expect(wrapper).toMatchSnapshot();
    });

    test('section', () => {
      const wrapper = shallow(<Container As={'section'} children={<div />} />);

      expect(wrapper).toMatchSnapshot();
    });
  });
});
