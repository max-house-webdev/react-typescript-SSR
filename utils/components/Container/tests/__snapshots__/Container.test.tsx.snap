// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> should render as div by default 1`] = `
<div>
  <div />
</div>
`;

exports[`<Container /> should render as header 1`] = `
<header>
  <div />
</header>
`;

exports[`<Container /> should render as section 1`] = `
<section>
  <div />
</section>
`;
